### DISCUSSION TITLE
IMU Sensor tracking

### DISCUSSION THREAD

**Discussion by santhi04** [PlusToolkit/PlusLib#1099] [Q&A]
Hi everyone

 I am using an IMU sensor with a development board, and I want to use Slicer 3D for image visualisation. Can anyone explain me what process should I follow for the hardware connection and visualisation, or should we record data using Slicer 3D? If we have to use PLUS how should we write the configuration file.
 
Thank you
Nikhila

**Comment by santhi04** [comment]
<PlusConfiguration version="2.1">
    <DataCollection StartupDelaySec="1.0">
        <DeviceSet Name="PlusServer: MPU6050 Sensor" Description="Broadcasting MPU6050 sensor data through OpenIGTLink" />
        <Device Id="TrackerDevice" Type="GenericSensor" ToolReferenceFrame="Tracker">
            <DataSources>
                <DataSource Type="Tool" Id="IMU" PortName="MPU6050Sensor" />
            </DataSources>
            <OutputChannels>
                <OutputChannel Id="TrackerStream">
                    <DataSource Id="IMU"/>
                </OutputChannel>
            </OutputChannels>
        </Device>
        
        <Device Id="NucleoDevice" Type="SerialDevice">
            <DataSources>
                <DataSource Type="Tool" Id="NucleoSerial" PortName="/dev/ttyACM0" BaudRate="115200" />
            </DataSources>
        </Device>
        
        <Connector Id="NucleoConnector">
            <InputChannel Id="NucleoStream">
                <DataSource Id="NucleoSerial"/>
            </InputChannel>
            <OutputChannel Id="TrackerStream">
                <DataSource Id="NucleoSerial"/>
            </OutputChannel>
        </Connector>
        <Device Id="CaptureDevice" Type="VirtualCapture" BaseFilename="RecordingTest.igs.mha" EnableCapturingOnStart="FALSE">
            <InputChannels>
                <InputChannel Id="TrackerStream" />
            </InputChannels>
        </Device>
    </DataCollection>
    <PlusOpenIGTLinkServer MaxNumberOfIgtlMessagesToSend="1" MaxTimeSpentWithProcessingMs="50" ListeningPort="18944" OutputChannelId="TrackerStream">
        <DefaultClientInfo>
            <MessageTypes>
                <Message Type="TRANSFORM" />
            </MessageTypes>
            <TransformNames>
                <Transform Name="IMUToTracker" />
            </TransformNames>
        </DefaultClientInfo>
    </PlusOpenIGTLinkServer>
</PlusConfiguration>


I am using MPU6050 with NUCLEOL476RG development board. Can I use this configuration file ?

**Comment by Sunderlandkyl** [comment]
If you can communicate over a serial connection, you could look at the GenericSerialDevice: http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceGenericSerial.html.

There are some tracker devices that extend the basic serial functionality: [MicrochipTracker ](http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceMicrochip.html),  [WitMotionTracker](http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceWitMotionTracker.html).
